lib/client/session.go

```diff
@@ -539,18 +544,23 @@ func (ns *NodeSession) watchSignals(shell io.Writer) {
 
 // pipeInOut launches two goroutines: one to pipe the local input into the remote shell,
 // and another to pipe the output of the remote shell into the local output
-func (ns *NodeSession) pipeInOut(shell io.ReadWriteCloser) {
+// func (ns *NodeSession) pipeInOut(shell io.ReadWriteCloser, wg *sync.WaitGroup) {
+func (ns *NodeSession) pipeInOut(shell io.ReadWriteCloser, wg *sync.WaitGroup) {
 	// copy from the remote shell to the local output
+	wg.Add(2)
 	go func() {
 		defer ns.closer.Close()
+		defer wg.Done()
```

Comment:
nit: put the `wg.Done` first, so that `ns.closer.Close()` is always guaranteed to have finished when `wg.Wait` returns